scripts/pipeline/orthology_benchmark.py

```diff
@@ -307,8 +307,51 @@ def run_orthology_tools(input_dir: str, orthofinder_params: str) -> None:
         raise RuntimeError(msg) from exc
 
 
-def prep_input_for_goc():
-    """Docstring"""
+def extract_orthologs(input_dir: str, species1: str, species2: str) -> List[Tuple[str, str]]:
+    """Returns a list of putative orthologous pairs inferred by OrthoFinder for a specified pair
+    of species.
+
+    If there are results for multiple OrthoFinder runs, the function uses those in the
+    most recently modified `Results_*` directory. This allows running the pipeline without user intervention.
+
+    Args:
+        input_dir: Path to the directory with OrthoFinder output.
+        species1: Name of one species of interest.
+        species2: Name of another species of interest.
+
+    Raises:
+        FileNotFoundError: If there is no `OrthoFinder/Results_*` directory in `input_dir`.
+
+    """
+    orthologs = []
+
+    try:
+        orthofinder_res = [f.path for f in os.scandir(os.path.join(input_dir, "OrthoFinder"))
+                           if f.is_dir() and f.name.startswith("Results")]
+    except FileNotFoundError as e:
+        raise FileNotFoundError("Could not find OrthoFinder output.") from e
```

Comment:
in this case I do not think you should catch the exception as the only think you are doing is adding is "Could not find OrthoFinder output" to the exception message and you raise it again. This is not crucial for a better understanding of the exception's default message. So I would either not catch the exception or just stop the script here (sys.exit() with the error message.